google-cloud-tools-plugin/src/com/google/cloud/tools/intellij/appengine/sdk/CloudSdkPanel.java

```diff
@@ -0,0 +1,127 @@
+/*
+ * Copyright 2016 Google Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.cloud.tools.intellij.appengine.sdk;
+
+import com.google.api.client.repackaged.com.google.common.annotations.VisibleForTesting;
+import com.google.cloud.tools.intellij.appengine.util.CloudSdkUtil;
+import com.google.cloud.tools.intellij.util.GctBundle;
+import com.google.cloud.tools.intellij.util.SystemEnvironmentProvider;
+
+import com.intellij.openapi.fileChooser.FileChooserDescriptorFactory;
+import com.intellij.openapi.options.ConfigurationException;
+import com.intellij.openapi.ui.TextFieldWithBrowseButton;
+import com.intellij.openapi.util.Comparing;
+import com.intellij.openapi.util.text.StringUtil;
+import com.intellij.ui.DocumentAdapter;
+import com.intellij.ui.JBColor;
+
+import org.jetbrains.annotations.NotNull;
+
+import javax.swing.JLabel;
+import javax.swing.JPanel;
+import javax.swing.event.DocumentEvent;
+
+/**
+ * Reusable panel for configuring the path to the Cloud SDK from various contexts.
+ */
+public class CloudSdkPanel {
+
+  private TextFieldWithBrowseButton cloudSdkDirectoryField;
+  private JLabel warningMessage;
+  private JPanel cloudSdkPanel;
+
+  public CloudSdkPanel(@NotNull CloudSdkService cloudSdkService) {
+    warningMessage.setVisible(false);
+
+    final String cloudSdkDirectoryPath
```

Comment:
I Guess what I'm saying is shouldn't we be detecting the path only if service.getCloudSdkHomePath == null? And shouldn't we be initializing the cloudSdkDirectoryField to service.getCloudSdkHomePath if it isn't null?